### PR TITLE
feat: folks can delete enclaves from the frontend

### PR DESCRIPTION
**Pull request by Peeeekay**
Users can delete enclave(s) from the frontend once this change get merged in!

### CODE REVIEW

File: engine/frontend/src/component/Home.js [hunk 1]
```diff
@@ -39,6 +38,16 @@ const Home = () => {
     const [enclaveLoading, setEnclaveLoading] = useState(false)
     const {appData, setAppData} = useAppContext()
 
+    const handleDeleteClick = (enclaveName) => {
+        console.log("name ", enclaveName)
+        const filteredEnclaves = enclaves.filter(enclave => enclave.name !== enclaveName)
+        const makeRequest = async () => {
+            await removeEnclave(appData.jwtToken, appData.apiHost, enclaveName)
```
Comment (adschwartz) [inline engine/frontend/src/component/Home.js]: @Peeeekay can we listen to the response and determine if the removal was successful or not?

File: engine/frontend/src/component/Home.js [hunk 2]
```diff
@@ -40,9 +40,14 @@ const Home = () => {
 
     const handleDeleteClick = (enclaveName) => {
         const makeRequest = async () => {
-            const filteredEnclaves = enclaves.filter(enclave => enclave.name !== enclaveName)
-            await removeEnclave(appData.jwtToken, appData.apiHost, enclaveName)
-            setEnclaves(filteredEnclaves)
+            try {
+                const filteredEnclaves = enclaves.filter(enclave => enclave.name !== enclaveName)
+                await removeEnclave(appData.jwtToken, appData.apiHost, enclaveName)
```
Comment (adschwartz) [inline engine/frontend/src/component/Home.js:45]: @Peeeekay doesn't removeEnclave return anything? THe endpoint must return an error code if it fails?

Comment (Peeeekay) [reply engine/frontend/src/component/Home.js:45]: No; the response failure normally are returned as errors - if you look at the proto schema it's just an empty response

File: engine/frontend/src/api/enclave.js [hunk 3]
```diff
@@ -19,6 +19,13 @@ export const getEnclavesFromKurtosis = async (token, apiHost) => {
     return []
 }
 
+export const removeEnclave = async (token, apiHost, enclaveName) => {
+    const response = await removeEnclaveFromEnclaveManager(enclaveName, token, apiHost)
+    const enclave = response.enclaveInfo;
```
Comment (adschwartz) [inline engine/frontend/src/api/enclave.js:24]: what does the `enclave` variable do?

**Review by adschwartz — APPROVED** [review]

File: engine/frontend/src/api/enclave.js (hunk 3 above)
Comment (Peeeekay) [reply engine/frontend/src/api/enclave.js:24]: just prints undefined, was using this for testing, forgot to remove it.

**Pull request merged**